### PR TITLE
Binder: Convert value back to representation write back to field when…

### PR DESCRIPTION
**Pull request by TatuLund**
Binder: Convert value back to representation write back to field when changed

See: https://vaadin.com/forum/thread/15931682 and https://stackoverflow.com/questions/44116251/vaadin-8-converter-behaves-different-than-vaadin-7-converter-doesnt-update-ui

---
This change is [Reviewable](https://reviewable.io/reviews/vaadin/framework/12132)

### CODE REVIEW

**Review by Ansku — APPROVED** [review]
:lgtm:

> Reviewed 1 of 2 files at r1, 1 of 1 files at r2.
> *[Reviewable](https://reviewable.io/reviews/vaadin/framework/12132#-:-MNabjiK1VWQh5MHfUIK:bnfp4nl)* status: :shipit: complete! all files reviewed, all discussions resolved

File: server/src/main/java/com/vaadin/data/Binder.java [hunk 1]
```diff
@@ -1285,7 +1285,10 @@ private void handleFieldValueChange(
                 result.ifOk(value -> {
                     setter.accept(bean, value);
                     if (value != null) {
-                        getField().setValue(convertToFieldType(value));
+                        FIELDVALUE converted = convertToFieldType(value);
+                        if (!field.getValue().equals(converted)) {
```
Comment (alexandru-prigoreanu) [inline server/src/main/java/com/vaadin/data/Binder.java:1289]: hello! this instruction raises NullPointerException when field.getValue() is null. should i open a new issue for it? thank you!

Comment (TatuLund) [reply server/src/main/java/com/vaadin/data/Binder.java:1289]: This is fixed in 8.12.4

Comment (alexandru-prigoreanu) [reply server/src/main/java/com/vaadin/data/Binder.java:1289]: thank you!

**Pull request merged**